tests/checks/mock/test_mapreduce.py

```diff
@@ -0,0 +1,231 @@
+        'mapreduce.job.reduce.task.elapsed_time.max': 123456
+    }
+
+    MAPREDUCE_REDUCE_TASK_METRIC_TAGS = [
+        'cluster_id:' + CLUSTER_ID,
+        'app_name:' + APP_NAME,
+        'job_name:' + JOB_NAME,
+        'user_name:' + USER_NAME,
+        'task_type:reduce'
+    ]
+
+    MAPREDUCE_JOB_COUNTER_METRIC_VALUES = {
+        'mapreduce.job.counter.total_counter_value.max': {'value': 0, 'tags': ['counter_name:file_bytes_read']},
+        'mapreduce.job.counter.map_counter_value.max': {'value': 1, 'tags': ['counter_name:file_bytes_read']},
+        'mapreduce.job.counter.reduce_counter_value.max': {'value': 2, 'tags': ['counter_name:file_bytes_read']},
+        'mapreduce.job.counter.total_counter_value.max': {'value': 3, 'tags': ['counter_name:file_bytes_written']},
+        'mapreduce.job.counter.map_counter_value.max': {'value': 4, 'tags': ['counter_name:file_bytes_written']},
+        'mapreduce.job.counter.reduce_counter_value.max': {'value': 5, 'tags': ['counter_name:file_bytes_written']},
+        'mapreduce.job.counter.total_counter_value.max': {'value': 9, 'tags': ['counter_name:map_output_records']},
+        'mapreduce.job.counter.map_counter_value.max': {'value': 10, 'tags': ['counter_name:map_output_records']},
+        'mapreduce.job.counter.reduce_counter_value.max': {'value': 11, 'tags': ['counter_name:map_output_records']},
+    }
+
+    MAPREDUCE_JOB_COUNTER_METRIC_TAGS = [
+        'cluster_id:' + CLUSTER_ID,
+        'app_name:' + APP_NAME,
+        'job_name:' + JOB_NAME,
+        'user_name:' + USER_NAME
+    ]
+
+    @mock.patch('requests.get', side_effect=requests_get_mock)
+    def test_check(self, mock_requests):
+        config = {
+            'instances': [self.MR_CONFIG],
+            'init_config': self.INIT_CONFIG
+        }
+
+        self.run_check(config)
+
+        # Check the MapReduce job metrics
+        for metric, value in self.MAPREDUCE_JOB_METRIC_VALUES.iteritems():
+            self.assertMetric(metric,
+                value=value,
+                tags=self.MAPREDUCE_JOB_METRIC_TAGS)
+
+        # Check the map task metrics
+        for metric, value in self.MAPREDUCE_MAP_TASK_METRIC_VALUES.iteritems():
+            self.assertMetric(metric,
+                value=value,
+                tags=self.MAPREDUCE_MAP_TASK_METRIC_TAGS)
+
+        # Check the reduce task metrics
+        for metric, value in self.MAPREDUCE_REDUCE_TASK_METRIC_VALUES.iteritems():
+            self.assertMetric(metric,
+                value=value,
+                tags=self.MAPREDUCE_REDUCE_TASK_METRIC_TAGS)
+
+        # Check the MapReduce job counter metrics
+        for metric, attributes in self.MAPREDUCE_JOB_COUNTER_METRIC_VALUES.iteritems():
+            tags = attributes['tags']
+            tags.extend(self.MAPREDUCE_JOB_COUNTER_METRIC_TAGS)
+            self.assertMetric(metric,
+                value=attributes['value'],
+                tags=tags)
```

Comment:
You can also add assertions on the service checks using `self.assertServiceCheckOK` for instance (see https://github.com/DataDog/dd-agent/blob/8e36e58ed13bdb735a6c75e2c7b21f8998457c6a/tests/checks/common.py#L484 for reference)